packages/firestore/src/local/memory_persistence.ts

```diff
@@ -61,6 +68,16 @@ export class MemoryPersistence implements Persistence {
     return Promise.resolve();
   }
 
+  tryBecomePrimary(): Promise<boolean> {
+    // All clients using memory persistence act as primary.
+    return Promise.resolve(true);
+  }
+
+  setPrimaryStateListener(primaryStateListener: PrimaryStateListener) {
+    // All clients using memory persistence act as primary.
+    this.queue.enqueue(() => primaryStateListener(true));
```

Comment:
If you keep the IndexedDb code that calls this with false on shutdown I'd suggest that you do the same for MemoryPersistence.  But really, I think we can probably just treat shutdown as a special case and not do the notification.